File: Daily Test/Java/Longest Line Length 15.7.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Map<Integer, List<Integer>> map = new HashMap<>();
        for (int i = 0; i < n; i++) {
            int x = sc.nextInt(), y = sc.nextInt();
            map.computeIfAbsent(y, k -> new ArrayList<>()).add(x);
        }
        int max = -1;
        for (List<Integer> list : map.values()) {
            if (list.size() >= 2) {
                Collections.sort(list);
                max = Math.max(max, list.get(list.size() - 1) - list.get(0));
            }
        }
        System.out.println(max);
    }
}

File: Daily Test/Python/Longest Line Length 15.7.2025.py
n = int(input())
a = [tuple(map(int, input().split())) for _ in range(n)]
from collections import defaultdict
b = defaultdict(list)
for c, d in a: b[d].append(c)
e = -1
for f in b.values():
    if len(f) >= 2: f.sort(); e = max(e, f[-1] - f[0])
print(e)